feat: add experiences data table

ccjjmartin opened the pull request:
## Purpose:
- feat: add experiences data table
- Staff listing data table and page
- CSU-390: Add partner experience details page

### Ticket(s)
[CSU-320](https://fourkitchens.clickup.com/t/36718269/CSU-320)
[CSU-383](https://fourkitchens.clickup.com/t/36718269/CSU-383)
[CSU-390](https://fourkitchens.clickup.com/t/36718269/CSU-390)

### Pull Request Deployment:
- Normal deployment process (`npm run rebuild` on local and ci on github)

### Functional Testing:
- [x] View: http://localhost:6006/?path=/story/components-data-table--generic
- [x] View: http://localhost:6006/?path=/story/components-data-table--experiences
- [x] View: http://localhost:6006/?path=/story/components-data-table--staff
- [x] View: http://localhost:6006/?path=/story/components-form--staff
- [x] View: http://localhost:6006/?path=/story/pages-generic--staff-table
- [x] View: http://localhost:6006/?path=/story/pages-generic--experience-table
- [x] View: http://localhost:6006/?path=/story/pages-partner-organization-experiences-experience-details--default

mcortes19 commented on src/components/02-components/data-table/data-table-experiences.tsx:
```diff
@@ -0,0 +1,132 @@
+        />,
+        <GridActionsCellItem
+          icon={<OpenInBrowserIcon />}
+          label="VIEW DETAILS"
```
We avoid capitalized words due screen readers, if any text should be capitalized that needs to be done using css
```suggestion
          label="View details"
```

mcortes19 commented on src/components/02-components/data-table/data-table-experiences.tsx:
```diff
@@ -0,0 +1,132 @@
+
+  function viewExperience(id: GridRowId) {
+    return () => {
+      window.location.href = '/organization/experiences/' + id;
```
We need to avoid `window.location.href` when navigating. Is there other way to use `Button Link` component instead of `GridActionsCellItem` for link actions? 

Also notice the url shouldn't be hardcoded here, it needs to be passed down from the FE app instead

mcortes19 commented on src/components/02-components/data-table/data-table-experiences.tsx:
```diff
@@ -0,0 +1,132 @@
+import React from 'react';
```
This is no longer needed, so get rid of it

mcortes19 commented on src/components/02-components/data-table/data-table-experiences.tsx:
```diff
@@ -0,0 +1,132 @@
+
+export type Row = {
+  id: GridRowId;
+  col1: string;
```
We need the ability to make rows optional. They are passed from the FE based on the GraphQL query and in some cases we need to do some tricks to pass Typescript checks: https://github.com/fourkitchens/s4-app-partners/actions/runs/5955540322/job/16154389142?pr=16
```suggestion
  col1?: string;
```

mcortes19 reviewed (changes requested):
@ccjjmartin please take a look at mobile styles for the table: 
![Screenshot 2023-08-23 at 11 33 43](https://github.com/fourkitchens/s4-shared-components/assets/11282049/66ab229f-71cb-4ba7-adbc-6acfe2ab2cde)
![Screenshot 2023-08-23 at 11 34 28](https://github.com/fourkitchens/s4-shared-components/assets/11282049/7c30c8af-e7a3-45ef-890d-5c0748f36b6d)

I also left some comments, feel free to reach out if you need id

ccjjmartin commented:
@mcortes19 Ready for re-review I have removed the filters button and removed the broken pager option "all" and swapped out the actions for buttons, I am unable to view the component in storybook after the button swap out but it works in the partners app.  If you have ideas on the `useHref() may be used only in the context of a <Router> component.` error let me know but multiple pages in storybook have that error (the ones with the buttons).

ccjjmartin commented:
Moving back to needs work, the href thing seems like an actual issue that probably won't be isolated to just my local, pushing up the changes requested from the Partner app into this component too.

mcortes19 reviewed (commented):
@ccjjmartin let's have a meeting to go over some implementation details here

mcortes19 commented on src/components/02-components/data-table/data-table-experiences.tsx:
```diff
@@ -0,0 +1,124 @@
+export default function DataTableExperiences({ initialRows }: DataTableProps) {
+  const theme = useTheme();
+
+  const columns: GridColDef<ExperienceRow>[] = [
```
Can we build the rows dynamically instead of hardcode them?

mcortes19 commented on src/components/04-pages/partners/organization/experiences/experiences.tsx:
```diff
@@ -0,0 +1,40 @@
+import Breadcrumbs from '../../../../01-elements/breadcrumbs/breadcrumbs';
+import DataTableExperiences, { ExperienceRow } from '../../../../02-components/data-table/data-table-experiences';
+
+export type PartnerExperiencePageProps = {
```
We need to define separate variables this Component accepts instead of passing an entire object

mcortes19 commented on src/components/04-pages/partners/organization/experiences/experiences.tsx:
```diff
@@ -0,0 +1,40 @@
+  };
+
+  return (
+    <div>
```
Use React fragment instead of an div for wrapping content
